src/components/Breadcrumb/Breadcrumb.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Breadcrumb as CarbonBreadcrumb,
+  OverflowMenu as CarbonOverflowMenu,
+} from 'carbon-components-react';
+import useResizeObserver from 'use-resize-observer';
+
+import BreadcrumbOverflowItem from './BreadcrumbOverflowItem';
+import './_breadcrumb.scss';
+
+const propTypes = {
+  /** Pass in the BreadcrumbItem's for your Breadcrumb */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  children: null,
+};
+
+const Breadcrumb = props => {
+  const { children, ...other } = props;
+  const [breadcrumbItemsWidth, setBreadcrumbItemsWidth] = useState([0]);
+  const [totalWidth, setTotalWidth] = useState(0);
+  const { ref: breadcrumbRef, width: containerWidth } = useResizeObserver({
+    useDefaults: false,
+  });
+  const [breadcrumbChildren, setBreadcrumbChildren] = useState(children);
+  const [overflowIndex, setOverflowIndex] = useState(0);
+
+  /** intial to get children total width */
+  useEffect(() => {
+    // measure child total width
+    const itemsWidth = [];
+    const breadcrumb = breadcrumbRef.current;
+    const breadcrumItems = breadcrumb.querySelectorAll('li.bx--breadcrumb-item');
+
+    if (breadcrumItems.length > 0) {
+      for (let i = 0; i <= breadcrumItems.length - 1; i += 1) {
+        itemsWidth.push(breadcrumItems[i].offsetWidth);
+      }
+      setBreadcrumbItemsWidth(itemsWidth);
+      const curTotalWidth = itemsWidth.reduce((acc, cur) => acc + cur);
+      setTotalWidth(curTotalWidth);
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  /** adjust overflow items number to responce container width  */
+  useEffect(() => {
+    let curTotalWidth = breadcrumbItemsWidth.reduce((acc, cur) => acc + cur);
+
+    if (containerWidth >= curTotalWidth) {
+      setBreadcrumbChildren(children);
+    } else {
+      for (let i = 1; i <= breadcrumbItemsWidth.length - 2; i += 1) {
+        curTotalWidth -= breadcrumbItemsWidth[i];
+        if (containerWidth >= curTotalWidth) {
+          setOverflowIndex(i);
+          break;
+        }
+      }
+      if (containerWidth < curTotalWidth) {
+        setOverflowIndex(-1);
+      }
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [containerWidth]);
```

Comment:
It's important that all dependent values are included here to prevent bugs. Was this intended? `breadcrumbItemsWidth` and `children` are both dependent values utilized in the function and should be included here.